### PR TITLE
Refactor flint log format

### PR DESCRIPTION
**Pull request by noCharger**
### Description

Standardize Flint log output format:

```
{
  "timestamp": "",
  "severityText": "INFO",
  "severityNumber": 2,
  "body": {
    "message": "",
    "statusCode": ""
  },
  "attributes": {
    "domainName": "",
    "clientId": "",
    "exception.type": "",
    "exception.message": "",
  }
}
```

### Usage V2
1. Emit log using `CustomLogging` class

```
@Override
  public boolean test(T result) throws Throwable {
    int statusCode = ((HttpResponse) result).getStatusLine().getStatusCode();
    logInfo("Checking if status code is retryable: " + statusCode);

    boolean isRetryable = retryableStatusCodes.contains(statusCode);
    logInfo("Status code " + statusCode + " check result: " + isRetryable);
    return isRetryable;
  }
```

2. Example log

```
24/03/08 01:34:46 INFO OpenSearchUpdater: Checking if index exists: flint_ql_sessions
24/03/08 01:34:46 INFO CustomLogging: {"timestamp":1709861686263,"severityText":"INFO","severityNumber":9,"body":{"message":"Checking if status code is retryable: 200"},"attributes":{"clientId":"UNKNOWN","domainName":"UNKNOWN"}}
24/03/08 01:34:46 INFO CustomLogging: {"timestamp":1709861686270,"severityText":"INFO","severityNumber":9,"body":{"message":"Status code 200 check result: false"},"attributes":{"clientId":"UNKNOWN","domainName":"UNKNOWN"}}
24/03/08 01:34:47 INFO CustomLogging: {"timestamp":1709861687144,"severityText":"INFO","severityNumber":9,"body":{"message":"Checking if status code is retryable: 200"},"attributes":{"clientId":"UNKNOWN","domainName":"UNKNOWN"}}
24/03/08 01:34:47 INFO CustomLogging: {"timestamp":1709861687144,"severityText":"INFO","severityNumber":9,"body":{"message":"Status code 200 check result: false"},"attributes":{"clientId":"UNKNOWN","domainName":"UNKNOWN"}}
```

3. Contributor insights
Screenshot 2024-03-07 at 6 00 36 PM

### Issues Resolved
https://github.com/opensearch-project/opensearch-spark/issues/235

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by YANG-DB** [comment]
Hi
I suggest we follow the OTEL convention for such events - see https://opentelemetry.io/docs/specs/semconv/general/logs/

**Comment by noCharger** [comment]
> Hi I suggest we follow the OTEL convention for such events - see https://opentelemetry.io/docs/specs/semconv/general/logs/

Could you please propose the log format? This link of General Logs Attributes is marked as Experimental

**Review by vamsi-amazon — APPROVED** [review]

File: flint-core/src/main/java/org/opensearch/flint/core/logging/CustomJsonLayout.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+/**
+ * CustomJsonLayout is a plugin for formatting log events as JSON strings.
+ *
+ * <p>The layout is designed to be used with OpenSearch Flint logging. It extracts environment-specific information,
+ * such as the cluster name, from the environment variable "FLINT_CLUSTER_NAME" and splits it into domain name and client ID.</p>
+ *
+ */
+@Plugin(name = "CustomJsonLayout", category = "Core", elementType = Layout.ELEMENT_TYPE, printObject = true)
+public class CustomJsonLayout extends AbstractStringLayout {
```
Comment (penghuo) [inline flint-core/src/main/java/org/opensearch/flint/core/logging/CustomJsonLayout.java:29]: 1. do we need custom json layout? can we leverage Log4j json layout? https://logging.apache.org/log4j/2.x/manual/json-template-layout.html
2. can we leverage Log4j env lookup for domain_name, client_id? https://logging.apache.org/log4j/2.x/manual/lookups.html#EnvironmentLookup

**Comment by noCharger** [comment]
> Hi I suggest we follow the OTEL convention for such events - see https://opentelemetry.io/docs/specs/semconv/general/logs/

Propose a new log format following OTEL convention 

```
{
  "timestamp": "",
  "severityText": "INFO",
  "body": {
    "message": "",
    "statusCode": ""
  },
  "attributes": {
    "domainName": "",
    "clientId": "",
    "exception.type": "",
    "exception.message": "",
    "exception.stacktrace": ""
  }
}
```

cc: @YANG-DB @penghuo @dai-chen @vamsi-amazon

**Comment by YANG-DB** [comment]
Nice !
Can we also calculate the [severityNumber](https://opentelemetry.io/docs/specs/otel/logs/data-model/#field-severitynumber) ?

**Comment by noCharger** [comment]
To prevent overriding log4j properties, create a mock log file using the default pattern layout
Screenshot 2024-03-07 at 10 23 27 AM

Screenshot 2024-03-07 at 10 26 01 AM

**Comment by noCharger** [comment]
@YANG-DB @penghuo  Resolved your comments. Please take another review. Thanks!

**Review by YANG-DB — APPROVED** [review]

**Pull request merged**